Microsoft.Toolkit.Uwp.UI.Controls.Core/ImageEx/ImageExBase.Source.cs

```diff
@@ -112,47 +116,38 @@ private async void SetSource(object source)
                 return;
             }
 
-            _uri = source as Uri;
-            if (_uri == null)
+            CurrentSourceUri = source as Uri;
+            if (CurrentSourceUri == null)
             {
                 var url = source as string ?? source.ToString();
-                if (!Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out _uri))
+                if (!Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out Uri uri))
                 {
                     VisualStateManager.GoToState(this, FailedState, true);
                     return;
                 }
+
+                CurrentSourceUri = uri;
             }
 
-            _isHttpSource = IsHttpUri(_uri);
-            if (!_isHttpSource && !_uri.IsAbsoluteUri)
+            if (!IsHttpUri(CurrentSourceUri) && !CurrentSourceUri.IsAbsoluteUri)
             {
-                _uri = new Uri("ms-appx:///" + _uri.OriginalString.TrimStart('/'));
+                CurrentSourceUri = new Uri("ms-appx:///" + CurrentSourceUri.OriginalString.TrimStart('/'));
             }
 
-            await LoadImageAsync(_uri);
+            await LoadImageAsync(CurrentSourceUri);
```

Comment:
This await call is done in an `async void` method, and the callee doesn't have a `try/catch` block - if there was an exception the whole process would likely crash. Is this intentional, or would we want to add a safety path there to instead just display nothing and raise the `ImageFailed` event and corresponding visual state?

I see that the various `try/catch` blocks are mentioned in the docs for the custom `LoadCachedImageAsync` instead, what if we used them here instead so that they'd automatically work for overridden methods too, but they'd also at the same time shield the control from other possible crashes in the default implementation?

Reply:
Done, we handle everything now. I did find that the original code was calling open/failed twice, as that's handled automatically by us attaching the source, so that should be all cleaned-up now too in the new commit.